### ISSUE TITLE
hostIP is no longer needed. All local connection should use either localhost or docker stack names. Neither public host IP nor DNS name should be used.

### ISSUE THREAD

**Issue by sunami09** [esnet/sense-rtmon#80]

**Comment by sunami09** [comment]
```bash
[root@ip-172-31-72-189 sense-rtmon]# curl http://pushgateway:9091/metrics
curl: (6) Could not resolve host: pushgateway; Unknown error
[root@ip-172-31-72-189 sense-rtmon]# docker ps
CONTAINER ID   IMAGE                                COMMAND                  CREATED          STATUS          PORTS      NAMES
52c4cb83d7b1   grafana/grafana-enterprise:latest    "/run.sh"                19 minutes ago   Up 19 minutes   3000/tcp   cloud_grafana.1.w1efq268kbyqkcgzlduf6pa9o
2fbef71901e7   prom/prometheus:v2.2.1               "/bin/prometheus --c…"   19 minutes ago   Up 19 minutes   9090/tcp   cloud_prometheus.1.zxgktry5m9g74grtqhrb9mq75
3cfda5d674ad   ricoberger/script_exporter:v2.16.0   "/sbin/tini -- /bin/…"   19 minutes ago   Up 19 minutes   9469/tcp   cloud_script_exporter.1.ecctvhecw90crty3l7427zzc0
a50c50e3450f   prom/pushgateway:latest              "/bin/pushgateway"       19 minutes ago   Up 19 minutes   9091/tcp   cloud_pushgateway.1.r5habqgl3d7v1bj5eye2utwqv
04a6f23d12e0   nginx:latest                         "/docker-entrypoint.…"   19 minutes ago   Up 19 minutes   80/tcp     cloud_nginx.1.b0evssdt12ulbj5bg5l735u5b



root@ip-172-31-72-189 sense-rtmon]# docker network ls
NETWORK ID     NAME                DRIVER    SCOPE
759fe8f6a780   bridge              bridge    local
4jnsy5e6gi1q   cloud_default       overlay   swarm
l79gg59o39rc   cloud_monitor-net   overlay   swarm
e5b6da7d544e   docker_gwbridge     bridge    local
b1e7138aac4b   host                host      local
hih5pc7r3ujn   ingress             overlay   swarm
dbb4aa3c1a9e   none                null      local
[root@ip-172-31-72-189 sense-rtmon]# docker network inspect cloud_monitor-net
[
    {
        "Name": "cloud_monitor-net",
        "Id": "l79gg59o39rcuy1rhhc7l5q9m",
        "Created": "2024-03-08T06:02:33.86821058Z",
        "Scope": "swarm",
        "Driver": "overlay",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "10.0.6.0/24",
                    "Gateway": "10.0.6.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "2fbef71901e74000398449894b0e5670672c871f22b70b7a7b732b30eaeb0004": {
                "Name": "cloud_prometheus.1.zxgktry5m9g74grtqhrb9mq75",
                "EndpointID": "1f3753427fc9ea6deaeef212a6f7b891608bfd80a969a474d9b5316442872d50",
                "MacAddress": "02:42:0a:00:06:03",
                "IPv4Address": "10.0.6.3/24",
                "IPv6Address": ""
            },
            "3cfda5d674ad5eed6513102b4b5e5033fac17cb7d34eef4aeb351f77e399a116": {
                "Name": "cloud_script_exporter.1.ecctvhecw90crty3l7427zzc0",
                "EndpointID": "33707398bd97527d89e0ecbdd413afa9e389a041acf5a70e2d9f0fb0ae0a6c12",
                "MacAddress": "02:42:0a:00:06:08",
                "IPv4Address": "10.0.6.8/24",
                "IPv6Address": ""
            },
            "a50c50e3450f54b9e20b26434d7f3d5d0b06c53675afdabc7deff6868ac290cb": {
                "Name": "cloud_pushgateway.1.r5habqgl3d7v1bj5eye2utwqv",
                "EndpointID": "11640793063b231554ddbe0754d830076aa0cc13d56ef7dd44843e83c7601b22",
                "MacAddress": "02:42:0a:00:06:06",
                "IPv4Address": "10.0.6.6/24",
                "IPv6Address": ""
            },
            "lb-cloud_monitor-net": {
                "Name": "cloud_monitor-net-endpoint",
                "EndpointID": "2954af246de4f6aa09154bf531e90da8f66ce278bafc8249e72db4022bea2e82",
                "MacAddress": "02:42:0a:00:06:04",
                "IPv4Address": "10.0.6.4/24",
                "IPv6Address": ""
            }
        },
        "Options": {
            "com.docker.network.driver.overlay.vxlanid_list": "4102"
        },
        "Labels": {
            "com.docker.stack.namespace": "cloud"
        },
        "Peers": [
            {
                "Name": "07e83f94d2d2",
                "IP": "172.31.72.189"
            }
        ]
    }
]

```

**Comment by sunami09** [comment]
```bash
#!/bin/bash 

        # check_arp_on
        if curl http://172.31.72.189:9091/metrics | grep '.*arp_state.*' | grep '.*flow="rtmon-d6a83a69-23d8-42aa-99b9-8c3bb31ad332".*' | grep '.*instance="T2_US_UCSD:k8s-igrok-02.calit2.optiputer.net".*'; then
            echo 't2_us_ucsd:k8s_igrok_02_calit2_optiputer_net_script_exporter_task1_rtmon_d6a83a69_23d8_42aa_99b9_8c3bb31ad332{host="t2_us_ucsd:k8s_igrok_02_calit2_optiputer_net"} 1' | curl --data-binary @- http://dev2.virnao.com:9091/metrics/job/single
        else
            echo 't2_us_ucsd:k8s_igrok_02_calit2_optiputer_net_script_exporter_task1_rtmon_d6a83a69_23d8_42aa_99b9_8c3bb31ad332{host="t2_us_ucsd:k8s_igrok_02_calit2_optiputer_net"} 0' | curl --data-binary @- http://dev2.virnao.com:9091/metrics/job/single
        fi
        
        # check_arp_contain_ip
        if curl http://172.31.72.189:9091/metrics | grep '.*arp_state.*' | grep '.*flow="rtmon-d6a83a69-23d8-42aa-99b9-8c3bb31ad332".*' | grep '.*instance="T2_US_UCSD:k8s-igrok-02.calit2.optiputer.net".*' | grep '.*IPaddress="10.251.85.190".*'; then
            echo 't2_us_ucsd:k8s_igrok_02_calit2_optiputer_net_script_exporter_task2_rtmon_d6a83a69_23d8_42aa_99b9_8c3bb31ad332{host="t2_us_ucsd:k8s_igrok_02_calit2_optiputer_net"} 1' | curl --data-binary @- http://dev2.virnao.com:9091/metrics/job/single
        else
            echo 't2_us_ucsd:k8s_igrok_02_calit2_optiputer_net_script_exporter_task2_rtmon_d6a83a69_23d8_42aa_99b9_8c3bb31ad332{host="t2_us_ucsd:k8s_igrok_02_calit2_optiputer_net"} 0' | curl --data-binary @- http://dev2.virnao.com:9091/metrics/job/single
        fi
        
    # check_snmp_on
    if curl http://172.31.72.189:9091/metrics | grep '.*ifHCInOctets.*' | grep '.*instance="T2_US_UCSD:s1".*' | grep '.*flow="rtmon-d6a83a69-23d8-42aa-99b9-8c3bb31ad332".*'; then
        echo 't2_us_ucsd:s1_script_exporter_task1_rtmon_d6a83a69_23d8_42aa_99b9_8c3bb31ad332{host="t2_us_ucsd:s1"} 1' | curl --data-binary @- http://dev2.virnao.com:9091/metrics/job/single
    else
        echo 't2_us_ucsd:s1_script_exporter_task1_rtmon_d6a83a69_23d8_42aa_99b9_8c3bb31ad332{host="t2_us_ucsd:s1"} 0' | curl --data-binary @- http://dev2.virnao.com:9091/metrics/job/single
    fi
    
    # check_snmp_on
    if curl http://172.31.72.189:9091/metrics | grep '.*ifHCInOctets.*' | grep '.*instance="calit2.optiputer.net:2020:prism-core".*' | grep '.*flow="rtmon-d6a83a69-23d8-42aa-99b9-8c3bb31ad332".*'; then
        echo 'calit2_optiputer_net:2020:prism_core_script_exporter_task1_rtmon_d6a83a69_23d8_42aa_99b9_8c3bb31ad332{host="calit2_optiputer_net:2020:prism_core"} 1' | curl --data-binary @- http://dev2.virnao.com:9091/metrics/job/single
    else
        echo 'calit2_optiputer_net:2020:prism_core_script_exporter_task1_rtmon_d6a83a69_23d8_42aa_99b9_8c3bb31ad332{host="calit2_optiputer_net:2020:prism_core"} 0' | curl --data-binary @- http://dev2.virnao.com:9091/metrics/job/single
    fi
    

```

**Comment by xi-yang** [comment]
The start.sh output gives this error, which may not harm but should be taken care of to avoid confusing users. @sunami09 
```
Traceback (most recent call last):
  File "prometheus.py", line 15, in <module>
    hostip = data['hostIP']
KeyError: 'hostIP'

```


Also piggybacking to this issue, for the final prompt / instructions of the start.sh output, we need to rephrase these below to make it more clear or maybe we no longer need some of these instructions.

```

!!    APT Key setup instruction (used for dynamic flow dashboard. ignore if API key already set up):
- setting -> API keys -> add key with Admin permission
- copy the API token value starting with 'Bearer ....'
- edit any files under /config_flow that are used
- replace 'CONFIG' in {grafanaAPIToken: 'CONFIG'} with the new API token

!!    Data source setup instruction:
- navigate to <ip_address/domain_name>:3000(or 443 if HTTPS enabled)
- login to Grafana with the default authentication (username: admin, password: admin)
- setting -> data source -> Prometheus -> URL -> Save & Test
- enter the IP address NOT DNS

```

**Comment by sunami09** [comment]
Fixed #88
